datumaro/components/validator.py

```diff
@@ -0,0 +1,951 @@
+                            bbox_label_stats = bbox_dist_by_label[label_name]
+
+                            _x, _y, _w, _h = ann.get_bbox()
+                            area = ann.get_area()
+                            ratio = _w / _h
+                            _short = _w if _w < _h else _h
+                            _long = _w if _w > _h else _h
+
+                            ann_bbox_info = {
+                                'x': _x,
+                                'y': _y,
+                                'width': _w,
+                                'height': _h,
+                                'area(wxh)': area,
+                                'ratio(w/h)': ratio,
+                                'short': _short,
+                                'long': _long,
+                            }
+
+                            for prop, val in ann_bbox_info.items():
+                                prop_stats = bbox_label_stats[prop]
+                                items_far_from_mean = \
+                                    prop_stats['items_far_from_mean']
+                                mean = prop_stats['mean']
+                                stdev = prop_stats['stdev']
+
+                                if val > mean + (_k * stdev) or \
+                                    val < mean - (_k * stdev):
+                                    anns_far_from_mean = \
+                                        items_far_from_mean.setdefault(
+                                            item.id, {})
+                                    anns_far_from_mean[ann.id] = val
+
+                            for attr, value in ann.attributes.items():
+                                if attr in valid_attrs:
+                                    bbox_attr_stats = \
+                                        bbox_dist_by_attr[label_name][attr]
+                                    bbox_val_stats = bbox_attr_stats[value]
+
+                                    for prop, val in ann_bbox_info.items():
+                                        prop_stats = bbox_val_stats[prop]
+                                        items_far_from_mean = \
+                                            prop_stats['items_far_from_mean']
+                                        mean = prop_stats['mean']
+                                        stdev = prop_stats['stdev']
+
+                                        if val > mean + (_k * stdev) or \
+                                            val < mean - (_k * stdev):
+                                            anns_far_from_mean = \
+                                                items_far_from_mean.setdefault(
+                                                    item.id, {})
+                                            anns_far_from_mean[ann.id] = val
+
+        return stats
+
+    def _check_missing_label_categories(self, stats):
+        validation_reports = []
+        description = "'LabelCategories(...)' should be defined" \
```

Comment:
Consider using and extending typed errors from `datumaro.components.errors`.

Avoid using implementation information in error messages, these class names are mostly useless for the potential customer.